File: Add Code Here/Swift/AddBinary.swift
class AddBinary {
    func addBinary(_ a: String, _ b: String) -> String {
        let a = Array(a), b = Array(b)
        var res = "", carry = 0, i = a.count - 1, j = b.count - 1

        while i >= 0 || j >= 0 || carry > 0 {
            var sum = carry

            if i >= 0 {
                sum += Int(String(a[i]))!
                i -= 1
            }
            if j >= 0 {
                sum += Int(String(b[j]))!
                j -= 1
            }

            res = "\(sum % 2)" + res
            carry = sum / 2
        }

        return res
    }
}

File: Add Code Here/C++/Add Binary.cpp
class Solution {
public:
    
    string addBinary(string a, string b) {
        string result;
        int i=a.length()-1;
        int j=b.length()-1;
        int carry=0;
        
        while(i>=0 || j>=0){
            int sum=carry;
            if(i>=0){
                sum+=a[i--]-'0';
            }
            if(j>=0){
                sum+=b[j--]-'0';
            }
            carry=sum>1?1:0;
            result+=to_string(sum%2);
        }
        if(carry){
            result+=to_string(carry);
        }
        reverse(result.begin(),result.end());
        
        return result;
    }
};